aya/src/maps/mod.rs

```diff
@@ -397,8 +508,8 @@ pub(crate) fn check_v_size<V>(map: &MapData) -> Result<(), MapError> {
 pub struct MapData {
     pub(crate) obj: obj::Map,
     pub(crate) fd: RawFd,
-    /// Indicates if this map has been pinned to bpffs
-    pub pinned: bool,
+    pub(crate) name: String,
+    pub(crate) paths: HashSet<PathBuf>,
```

Comment:
We shouldn't track paths here either. See #789.
It's fragile because "things that are not aya ™️ " can interact with the filesystem rendering the check pointless.
BPF_OBJ_PIN returns -EEXISTS in the case of pinning to a location that already exists... and there are already plenty of ways to check if a path exists before pinning.